### PR TITLE
Added the ability to perform detailed routing with different channel widths

### PR DESCRIPTION
**Pull request by heinerb1**
#### Description
I modified the init_chan function to prevent odd channel widths from being selected during task and regression test runs. In addition, I modified the binary_search_place_and_route function to pass directionality information to the init_chan function so that bidirectional graphs can still choose odd channel widths. Since I had to remove the VTR fatal error in CheckSetup.cpp to allow detailed routing, I also had to remove a function parameter in the CheckSetup function that was used only in the section that I removed.

#### Related Issue
#1768 

#### Motivation and Context
This change is necessary to create an approximate capture of Xilinx FPGA architectures in the VTR architecture description language, and for any other FPGA architectures that utilize different channel widths in the horizontal and vertical directions. This change allows you to utilize the existing VTR global routing distributions to perform detailed routing on FPGAs with different channel widths in the horizontal and vertical directions. Currently, these distributions can only be used for the global route type.

#### How Has This Been Tested?
I ran several individual vpr runs and tasks to test my changes. In addition, I ran the vtr_reg_basic and vtr_reg_strong regression tests. I am using the latest version of VTR and performed a clean build before running these tests. All of the tests ran successfully and passed the QoR test except for the QoR test for strong_global_nonuniform. This failed with the error message "x_gaussian_y_uniform.xml/stereovision3.v/common min_chan_width relative value 1.5 outside of range [0.25,1.3] and not equal to golden value: 12.0" However, I was wondering if it is important to pass this test since it is using the less common global route mode instead of the detailed routing mode? Other than this one issue, the code seems to have no adverse affects on other areas of the code. I would appreciate it if someone could let me know if I should run additional tests or recommend additional modifications to my changes.

#### Types of changes
- [x] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Checklist:
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

**Comment by heinerb1** [comment]
@vaughnbetz I made all of the changes that you requested and it should now pass all CI checks.

File: vpr/src/base/place_and_route.cpp [hunk 1]
```diff
@@ -452,6 +457,26 @@ t_chan_width init_chan(int cfactor, t_chan_width_dist chan_width_dist) {
     return chan_width;
 }
 
+/**
+ * @brief Computes the channel width and adjusts it to be an an even number if unidirectional 
+ *        since unidirectional graphs need to have paired wires.
+ * 
+ *   @param cfactor                 Channel width factor.
```
Comment (vaughnbetz) [inline vpr/src/base/place_and_route.cpp:464]: Could explain what channel width factor is: 
cfactor --> Channel width factor: multiplier on the channel width distribution (i.e. this is usually the number of tracks in the widest channel).

**Review by vaughnbetz — APPROVED** [review]
Looks good; thanks.  I just have one minor suggestion for an enhanced comment.

File: vpr/src/base/place_and_route.cpp (hunk 1 above)
Comment (heinerb1) [reply vpr/src/base/place_and_route.cpp]: Sure thing. I just updated it to include your description.

**Comment by vaughnbetz** [comment]
Looks like the kokoro tests weren't running as Brandon wasn't in the vtr org.  Just sent 
@heinerb1 an invite to vtr, and force started kokoro.

**Pull request merged**